TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -506,7 +506,54 @@ public static DataDefinition CheckVariable(Node node, StorageArea storageArea, b
                         node.GetEnclosingProgramOrFunctionNode().SetFlag(Node.Flag.UseGlobalStorage, true);
                     }
                 }
-                
+
+                if (!isReadStorageArea && node.SymbolTable.CurrentScope == SymbolTable.Scope.Function)
+                {
+                    FunctionDeclaration functionDeclaration = node.GetParent<Nodes.FunctionDeclaration>();
+                    if (functionDeclaration != null)
+                    {
+                        // search node variable in function declaration
+                        var variables = functionDeclaration.SymbolTable.GetVariablesExplicitWithQualifiedName(
+                            area.SymbolReference != null
+                                ? area.SymbolReference.URI
+                                : new URI(area.ToString()), null);
+                        var functionVariables = variables.Select(v => v.Value).ToList();
+                        if (functionVariables.Any())
+                        {
+                            // node variable is in function declaration: check if input parameter
+                            string name = string.Empty;
+                            var qualifiedName = functionVariables.First().QualifiedName;
+                            bool inputIsModified = false;
+                            foreach (var param in functionDeclaration.Profile.InputParameters)
```

Comment:
You don't need to iterate over InputParamteters.

You can check if dataDefinitionFound or dataDefinitionPath.CurrentDataDefinition are of type `ParameterDescription`
If so, property PassingType will tell you if it's an input parameter.